From 2227d3e6e114ed96382997ed9583e8833eb8e141 Mon Sep 17 00:00:00 2001
From: Ermine Jose <quic_erminejo@quicinc.com>
Date: Tue, 21 May 2024 16:58:27 +0530
Subject: [PATCH] include: override package configuration

when two package configuration with same name exist,
override the openwrt configuration. For example
the default configuration for busybox is taken from
package/utils/busybox, this change helps to override
with openwrt-patches/package/utils/busybox when
make defconfig is issued.

Signed-off-by: Ermine Jose <quic_erminejo@quicinc.com>
Change-Id: I8eda4e3d685fcce8a5f70af191978cd13b9bf3f3
(cherry picked from commit ef696085482347e660f9ebd6e39f77c6bce3530c)
---
 include/package-dumpinfo.mk | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/include/package-dumpinfo.mk b/include/package-dumpinfo.mk
index e180c770e3df..f97adfd74778 100644
--- a/include/package-dumpinfo.mk
+++ b/include/package-dumpinfo.mk
@@ -13,7 +13,10 @@ $(if $(PKG_BUILD_DEPENDS),Build-Depends: $(PKG_BUILD_DEPENDS)
 
 endef
 
+FindPackage ?=$(strip $(shell find $(TOPDIR)/openwrt-patches -name $(1) 2>/dev/null))
+
 define Dumpinfo/Package
+$(eval -include $(wildcard $(call FindPackage, $(1))/$(1).mk))
 $(info $(SOURCE_INFO)Package: $(1)
 $(if $(MENU),Menu: $(MENU)
 )$(if $(SUBMENU),Submenu: $(SUBMENU)
@@ -46,6 +49,7 @@ $(if $(KCONFIG),Kernel-Config: $(KCONFIG)
 )Description: $(if $(Package/$(1)/description),$(Package/$(1)/description),$(TITLE))
 @@
 $(if $(Package/$(1)/config),Config:
+$(Package/$(1)/overrideconfig)
 $(Package/$(1)/config)
 @@
 ))
-- 
2.34.1

